verifiers/internal/gcb/provenance.go

```diff
@@ -68,7 +53,7 @@ type gloudProvenance struct {
 type Provenance struct {
 	gcloudProv              *gloudProvenance
 	verifiedProvenance      *provenance
-	verifiedIntotoStatement *v01IntotoStatement
+	verifiedIntotoStatement *iface.Statement
```

Comment:
This should definitely not be a pointer to an interface (interface references are always pointers).
```suggestion
	verifiedIntotoStatement iface.Statement
```

Reply:
I am not sure this works, because we're testing that the point is nil in the code (this pointer is set only when we found a verified statement https://github.com/slsa-framework/slsa-verifier/blob/main/verifiers/internal/gcb/provenance.go#L541). 

Or maybe you're suggesting we replace the nil checks by an "empty interface" check?

Reply:
Since `iface.Statement` is an interface any reference is automatically a pointer so you can assign `nil` and check for it. You should almost never need a pointer to an interface. `*iface.Statement` is a pointer to a pointer.

Reply:
if `interface == nil` was not compiling when I tried, I think. I'll double check